src/linear_algebra/dual.lean

```diff
@@ -133,37 +128,37 @@ begin
   exact disjoint_bot_right
 end
 
-def to_dual_equiv [fintype B] : V ≃ₗ[K] (dual K V) :=
+def to_dual_equiv [fintype ι] : V ≃ₗ[K] (dual K V) :=
 linear_equiv.of_bijective h.to_dual h.to_dual_ker h.to_dual_range
 
-theorem dual_basis_is_basis [fintype B] : is_basis K h.dual_basis :=
+theorem dual_basis_is_basis [fintype ι] : is_basis K h.dual_basis :=
 h.to_dual_equiv.is_basis h
 
-@[simp] lemma to_dual_to_dual [fintype B] :
+@[simp] lemma to_dual_to_dual [decidable_eq (dual K (dual K V))] [fintype ι] :
   (h.dual_basis_is_basis.to_dual).comp h.to_dual = eval K V :=
 begin
-  apply h.ext,
-  intros b hb,
-  apply h.dual_basis_is_basis.ext,
-  intros c hc,
+  apply @is_basis.ext _ _ _ _ _ _ _ _ (classical.dec_eq (dual K (dual K V))) _ _ _ _ _ _ _ h,
```

Comment:
you should be able to use the `classical` tactic, then apply this normally

Reply:
Sorry, I can't make `classical` work here.